### PR TITLE
CompatibleDevice: Add tests

### PR DESCRIPTION
**Pull request by mlahargou**
This is a basic test to ensure that the base component renders correctly as well as the component renders correctly when the number of variants exceeds the limit. In that case, the component should display the `And 3 other models...` text. In this case, I wanted to generalize it a bit more so I only included `other models...` in the query text.

QA:
--
Tests should pass.

Connects: https://github.com/iFixit/react-commerce/issues/1143

### CODE REVIEW

**Comment by batbattur** [comment]
dev_block 🚒 looks like the type check CI check is failing

**Comment by mlahargou** [comment]
un_dev_block 📱 

I'd like a little advice on when/when not to use snapshot testing.

**Comment by jordycosta** [comment]
QA 🟢 

CI checks are passing ✔️ 
[View **Jest** test report](https://github.com/iFixit/react-commerce/actions/runs/3699281911/jobs/6266478105)

```
frontend test: PASS tests/jest/tests/ProductListItem.test.tsx
frontend test: PASS tests/jest/tests/CompatibleDevice.test.tsx
frontend test: Test Suites: 3 passed, 3 total
frontend test: Tests:       6 skipped, 6 passed, 12 total
frontend test: Snapshots:   3 passed, 3 total
frontend test: Time:        15.343 s
frontend test: Ran all test suites.
frontend test: Done
```

File: frontend/tests/jest/tests/CompatibleDevice.test.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+      let product = getMockProduct();
+      let device = product!.compatibility!.devices[0];
+
+      // @ts-ignore
+      const { getByText, asFragment } = renderWithAppContext(
+         <CompatibleDevice device={device} />
+      );
+
+      // getByText throws an error if it can't find the text
+      (expect(() => getByText(/other models.../i)) as any).toThrow();
```
Comment (batbattur) [inline frontend/tests/jest/tests/CompatibleDevice.test.tsx:16]:
```suggestion
      (expect(screen.queryByText(/other models.../i)) as any).not.toBeInTheDocument();
```

**Comment by mlahargou** [comment]
Great suggestions @batbattur ! un_deploy_block 📱

**Review by batbattur — APPROVED** [review]
CR 🚛

**Pull request merged**